src/Elewant/Herding/DomainModel/Breed/BreedCollection.php

```diff
@@ -91,7 +101,7 @@ public function equals(BreedCollection $otherCollection): bool
     {
         $diff = array_diff($this->breeds, $otherCollection->breeds);
 
-        return empty($diff) && count($this->breeds) === count($otherCollection->breeds);
+        return !$diff && count($this->breeds) === count($otherCollection->breeds);
```

Comment:
`empty($diff)` has the same behavior as `!isset($diff) || !$diff` (or `isset($diff) === false || $diff == false`). Since we _know_ `isset($diff)` will return `true`, because `$diff` is defined in the line before the `if`, that `!isset($diff)` part is irrelevant. What remains then, is `!$diff`.

I can follow that `$diff` is an array (or list), and we want to test "is the list empty?", then `if (empty($list))` makes sense. But IMHO we're not speaking English here, but we're speaking PHP. So, if we really want to make "is the diff empty?" explicit in PHP, it would look something like `count($diff) === 0`. I think it's important to speak English when naming things (variables, methods, etc), but the actual implementation remains PHP.

Personally `if ($var)` or `if (!$var)` (i.e. does it evaluate to true or false) is fine. I don't really need to go as far as `if ($var === true)` or `if ($var === false)` (or `count($diff) === 0`). But hey! It's a matter of taste / personal preference, so I'm open to alternatives :slightly_smiling_face:

Reply:
For me, it's a matter of exactly 0 bugs can come from the previous implementation with `empty`. As with `!diff`. So *shrug* and move on.